src/core/metrics.cc

```diff
@@ -343,6 +343,27 @@ instance_id_type shard() {
     return sstring("0");
 }
 
+void
+impl::set_metric_families_to_replicate(
+        std::unordered_set<replicated_family_spec> metric_families_to_replicate) {
+    // Remove replicas for all metric families that are not
+    // present in the new set.
+    for (const auto& spec: _metric_families_to_replicate) {
+        if (metric_families_to_replicate.contains(spec)) {
+            continue;
+        }
+
+        remove_metric_replica_family(spec);
+    }
+
+    // Replicate the specified metric families.
+    for (const auto& spec: metric_families_to_replicate) {
+        replicate_metric_family(spec);
```

Comment:
Does this work well if the metric is already registered?

Reply:
No. You'd get a `double_registration` exception thrown, which goes against the semantics of a `set` function. Will fix.

Reply:
Fixed in the latest force push and added a unit test too.